### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by jcrosnierw6d**

### CODE REVIEW

File: charts/ci-operator/templates/steps/standard.yaml [hunk 1]
```diff
@@ -86,6 +86,460 @@ step:
...
+  - name: CLIENT_HTTP_REPOSITORY_URL
+    description: defines the http(s) repository
+    type: string
+step:
+  name: static-test-gitleakspathscan
+  image: "zricethezav/gitleaks:latest"
+  script: |
+    mkdir -p $(workspaces.source.path)/tests/report-gitleaks
+    cd $(workspaces.source.path)/tests/
+    git clone $(params.CLIENT_HTTP_REPOSITORY_URL) repo-gitleaks
```
Comment (david972) [inline charts/ci-operator/templates/steps/standard.yaml:109]: no ! the git clone is already done by another step
and available in `$(resources.inputs.source.path)`

File: charts/ci-operator/templates/steps/standard.yaml [hunk 2]
```diff
@@ -86,6 +86,460 @@ step:
...
 ---
+apiVersion: ci.w6d.io/v1alpha1
+kind: Step
+metadata:
+  name: static-test-gitleakspathscan
+  labels:
+  {{- include "ci-operator.labels" . | nindent 4 }}
+  annotations:
+    ci.w6d.io/kind: generic
+    ci.w6d.io/task: static-test
```
Comment (david972) [inline charts/ci-operator/templates/steps/standard.yaml:97]: should be `git-leaks`

File: charts/ci-operator/templates/steps/standard.yaml [hunk 3]
```diff
@@ -86,6 +86,460 @@ step:
   script: |
     gitleaks -v --pretty --repo-path $(resources.inputs.source.path) --commit-from=$$(params.commitFrom) --commit-to=$(params.commitBefore) --branch=$(params.branch) --report gitleaks-report.json
 ---
+apiVersion: ci.w6d.io/v1alpha1
+kind: Step
+metadata:
+  name: static-test-gitleakspathscan
```
Comment (david972) [inline charts/ci-operator/templates/steps/standard.yaml:92]: may be `git-leaks-standard`

File: charts/ci-operator/templates/steps/standard.yaml [hunk 4]
```diff
@@ -86,6 +86,460 @@ step:
...
+    description: defines the http(s) repository
+    type: string
+step:
+  name: static-test-gitleakspathscan
+  image: "zricethezav/gitleaks:latest"
+  script: |
+    mkdir -p $(workspaces.source.path)/tests/report-gitleaks
+    cd $(workspaces.source.path)/tests/
+    git clone $(params.CLIENT_HTTP_REPOSITORY_URL) repo-gitleaks
+    gitleaks --path="repo-gitleaks/" -v --report=$(workspaces.source.path)/tests/report-gitleaks/report-pathsecretsscan.json
```
Comment (david972) [inline charts/ci-operator/templates/steps/standard.yaml:110]: the report should not be in the source but `$(workspaces.artifacts.path)`

File: charts/ci-operator/templates/steps/standard.yaml [hunk 5]
```diff
@@ -86,6 +86,460 @@ step:
...
+params:
+  - name: CLIENT_HTTP_REPOSITORY_URL
+    description: defines the http(s) repository
+    type: string
+step:
+  name: static-test-gitleakspathscan
+  image: "zricethezav/gitleaks:latest"
+  script: |
+    mkdir -p $(workspaces.source.path)/tests/report-gitleaks
+    cd $(workspaces.source.path)/tests/
```
Comment (david972) [inline charts/ci-operator/templates/steps/standard.yaml:108]: the report should not be in the source but $(workspaces.artifacts.path)

File: charts/ci-operator/templates/steps/standard.yaml [hunk 6]
```diff
@@ -86,6 +86,460 @@ step:
...
+---
+apiVersion: ci.w6d.io/v1alpha1
+kind: Step
+metadata:
+  name: static-test-codecov
+  labels:
+  {{- include "ci-operator.labels" . | nindent 4 }}
+  annotations:
+    ci.w6d.io/kind: generic
+    ci.w6d.io/task: static-test
```
Comment (david972) [inline charts/ci-operator/templates/steps/standard.yaml:120]: should be `codecov`

File: charts/ci-operator/templates/steps/standard.yaml [hunk 7]
```diff
@@ -86,6 +86,460 @@ step:
...
+  script: |
+    mkdir -p $(workspaces.source.path)/tests/report-gitleaks
+    cd $(workspaces.source.path)/tests/
+    git clone $(params.CLIENT_HTTP_REPOSITORY_URL) repo-gitleaks
+    gitleaks --path="repo-gitleaks/" -v --report=$(workspaces.source.path)/tests/report-gitleaks/report-pathsecretsscan.json
+---
+apiVersion: ci.w6d.io/v1alpha1
+kind: Step
+metadata:
+  name: static-test-codecov
```
Comment (david972) [inline charts/ci-operator/templates/steps/standard.yaml:115]: should be `codecov-standard`

File: charts/ci-operator/templates/steps/standard.yaml [hunk 8]
```diff
@@ -86,6 +86,460 @@ step:
...
+  labels:
+  {{- include "ci-operator.labels" . | nindent 4 }}
+  annotations:
+    ci.w6d.io/kind: generic
+    ci.w6d.io/task: static-test
+    ci.w6d.io/order: "0"
+params:
+  - name: INTERNAL_SONAR_URL
+    type: string
+    default: "https://sonar.staging.w6d.io"
```
Comment (david972) [inline charts/ci-operator/templates/steps/standard.yaml:152]: no

File: charts/ci-operator/templates/steps/standard.yaml [hunk 9]
```diff
@@ -86,6 +86,460 @@ step:
...
+  - name: CLIENT_SONAR_PROJECTNAME
+    type: string
+step:
+  name: static-test-sonarqube
+  image:
+    name: "sonarsource/sonar-scanner-cli:latest"
+    entrypoint: [""]
```
Comment (david972) [inline charts/ci-operator/templates/steps/standard.yaml:159]: are you sure about this ?

File: charts/ci-operator/templates/steps/standard.yaml [hunk 10]
```diff
@@ -86,6 +86,460 @@ step:
...
+  variables:
+    SONAR_USER_HOME: "${CI_PROJECT_DIR}/.sonar"  # Defines the location of the analysis task cache
+    GIT_DEPTH: "0"  # Tells git to fetch all the branches of the project, required by the analysis task
```
Comment (david972) [inline charts/ci-operator/templates/steps/standard.yaml:162]: does not work in tekton

File: charts/ci-operator/templates/steps/standard.yaml [hunk 11]
```diff
@@ -86,6 +86,460 @@ step:
...
+  cache:
+    key: "${CI_JOB_NAME}"
+    paths:
+      - .sonar/cache
```
Comment (david972) [inline charts/ci-operator/templates/steps/standard.yaml:166]: this neither

File: charts/ci-operator/templates/steps/standard.yaml [hunk 12]
```diff
@@ -86,6 +86,460 @@ step:
...
+  script: |
+    mkdir -p $(workspaces.source.path)/tests/report-codecov
+    cd $(workspaces.source.path)/tests/
+    cat <<EOF >sonar.properties
+    sonar.host.url=$(params.INTERNAL_SONAR_URL)
+    sonar.login=cd6f53e6ad5e3f6d7074f28fb1bb1eecabd9f893
+    sonar.projectKey=sonar:w6d:deploymgt
```
Comment (david972) [inline charts/ci-operator/templates/steps/standard.yaml:173]: to put in variable

File: charts/ci-operator/templates/steps/standard.yaml [hunk 13]
```diff
@@ -86,6 +86,460 @@ step:
...
+  cache:
+    key: "${CI_JOB_NAME}"
+    paths:
+      - .sonar/cache
+  script: |
+    mkdir -p $(workspaces.source.path)/tests/report-codecov
+    cd $(workspaces.source.path)/tests/
+    cat <<EOF >sonar.properties
+    sonar.host.url=$(params.INTERNAL_SONAR_URL)
+    sonar.login=cd6f53e6ad5e3f6d7074f28fb1bb1eecabd9f893
```
Comment (david972) [inline charts/ci-operator/templates/steps/standard.yaml:172]: seriously ? now I have to change this

File: charts/ci-operator/templates/steps/standard.yaml [hunk 14]
```diff
@@ -86,6 +86,460 @@ step:
...
+  - name: CLIENT_DOCKER_USER
+    description: defines the username credential for acces to the registry
+    type: string
+  - name: CLIENT_DOCKER_PASSWORD
+    description: defines the password for acces to the registry
+    type: string
+  - name: INTERNAL_CLAIR_URL
+    description: defines the clair url and the port entry of the cluster w6d (example IP:PORT).
+    type: string
+    default: "100.66.104.108:6060"
```
Comment (david972) [inline charts/ci-operator/templates/steps/standard.yaml:210]: what is it ?

File: charts/ci-operator/templates/steps/standard.yaml [hunk 15]
```diff
@@ -86,6 +86,460 @@ step:
...
+  - name: CLIENT_IMAGE_URL
+    description: defines the http(s) registry url adress for get the image
+    type: string
+  - name: CLIENT_DOCKER_USER
+    description: defines the username credential for acces to the registry
+    type: string
+  - name: CLIENT_DOCKER_PASSWORD
+    description: defines the password for acces to the registry
+    type: string
+  - name: INTERNAL_CLAIR_URL
```
Comment (david972) [inline charts/ci-operator/templates/steps/standard.yaml:207]: `CLAIR_URL` is enough

File: charts/ci-operator/templates/steps/standard.yaml [hunk 16]
```diff
@@ -86,6 +86,460 @@ step:
...
+    description: defines the http(s) target ip or hostname
+    type: string
+step:
+  name: static-test-owaspbaseline
+  image: "owasp/zap2docker-stable"
+  script: |
+    mkdir -p $(workspaces.source.path)/tests/report-owasp
+    cd $(workspaces.source.path)/tests/
+    mkdir -p /zap/wrk
+    zap-baseline.py -t $(params.CLIENT_HOST_URL) -J report-baseline.json -d
```
Comment (david972) [inline charts/ci-operator/templates/steps/standard.yaml:243]: `TARGET_HOST`

File: charts/ci-operator/templates/steps/standard.yaml [hunk 17]
```diff
@@ -86,6 +86,460 @@ step:
...
+    git clone https://gitlab.com/jasperdenkers/lunatech-demo-ci-scala-sbt.git sbtapp
+    cd sbtapp
+    sbt clean test
+    if [[ "$(params.COVERAGE)" == "true" ]] ; then
+          sbt clean coverage test coverageReport
+    fi
+    cd target
+    zip -r ../../sbt/target.zip *
+    cd ../../sbt
+    ls target.zip
```
Comment (david972) [inline charts/ci-operator/templates/steps/standard.yaml:478]: hmmm
the project will be already set ready

File: charts/ci-operator/templates/steps/standard.yaml [hunk 18]
```diff
@@ -86,6 +86,460 @@ step:
...
+step:
+  name: build-quasar
+  image: "node"
+  script: |
+    mkdir -p $(workspaces.source.path)/builds/quasar
+    cd $(workspaces.source.path)/builds
+    apt update
+    apt -y install git zip
+    npm i -g @quasar/cli
+    git clone https://github.com/quasarframework/quasar-ui-qmediaplayer.git quasarapp
```
Comment (david972) [inline charts/ci-operator/templates/steps/standard.yaml:520]: the repository should already be a quasar project

Comment (jcrosnierw6d) [reply charts/ci-operator/templates/steps/standard.yaml]: done

File: charts/ci-operator/templates/steps/standard.yaml (hunk 17 above)
Comment (jcrosnierw6d) [reply charts/ci-operator/templates/steps/standard.yaml]: done

File: charts/ci-operator/templates/steps/standard.yaml (hunk 16 above)
Comment (jcrosnierw6d) [reply charts/ci-operator/templates/steps/standard.yaml:236]: done

File: charts/ci-operator/templates/steps/standard.yaml (hunk 15 above)
Comment (jcrosnierw6d) [reply charts/ci-operator/templates/steps/standard.yaml]: done

File: charts/ci-operator/templates/steps/standard.yaml (hunk 14 above)
Comment (jcrosnierw6d) [reply charts/ci-operator/templates/steps/standard.yaml]: done

File: charts/ci-operator/templates/steps/standard.yaml (hunk 5 above)
Comment (jcrosnierw6d) [reply charts/ci-operator/templates/steps/standard.yaml]: done

File: charts/ci-operator/templates/steps/standard.yaml (hunk 4 above)
Comment (jcrosnierw6d) [reply charts/ci-operator/templates/steps/standard.yaml]: done

File: charts/ci-operator/templates/steps/standard.yaml (hunk 3 above)
Comment (jcrosnierw6d) [reply charts/ci-operator/templates/steps/standard.yaml]: done

File: charts/ci-operator/templates/steps/standard.yaml (hunk 2 above)
Comment (jcrosnierw6d) [reply charts/ci-operator/templates/steps/standard.yaml]: done

File: charts/ci-operator/templates/steps/standard.yaml (hunk 1 above)
Comment (jcrosnierw6d) [reply charts/ci-operator/templates/steps/standard.yaml]: done

File: charts/ci-operator/templates/steps/standard.yaml (hunk 11 above)
Comment (jcrosnierw6d) [reply charts/ci-operator/templates/steps/standard.yaml:163]: done

File: charts/ci-operator/templates/steps/standard.yaml [hunk 19]
```diff
@@ -86,6 +86,300 @@ step:
...
+    default: "https://codecov.io/bash"
+  - name: CODECOV_TOKEN
+    description: define by client defines the token (found by the user)
+    type: string
+step:
+  name: code-cov-standard
+  image: "bash"
+  script: |
+    mkdir -p $(workspaces.artifacts.path)/tests/report-codecov
+    echo y | apk add --no-cache curl
```
Comment (david972) [inline charts/ci-operator/templates/steps/standard.yaml:129]: as running in non root not available in CI

File: charts/ci-operator/templates/steps/standard.yaml [hunk 20]
```diff
@@ -86,6 +86,300 @@ step:
...
+---
+apiVersion: ci.w6d.io/v1alpha1
+kind: Step
+metadata:
+  name: owasp-baseline-standard
+  labels:
+  {{- include "ci-operator.labels" . | nindent 4 }}
+  annotations:
+    ci.w6d.io/kind: generic
+    ci.w6d.io/task: owasp-baseline
```
Comment (david972) [inline charts/ci-operator/templates/steps/standard.yaml:159]: set the same task for owasp steps

File: charts/ci-operator/templates/steps/standard.yaml [hunk 21]
```diff
@@ -86,6 +86,300 @@ step:
...
+---
+apiVersion: ci.w6d.io/v1alpha1
+kind: Step
+metadata:
+  name: nmap-topports100-standard
+  labels:
+  {{- include "ci-operator.labels" . | nindent 4 }}
+  annotations:
+    ci.w6d.io/kind: generic
+    ci.w6d.io/task: nmap-topports100
```
Comment (david972) [inline charts/ci-operator/templates/steps/standard.yaml:216]: set the same task for the nmap step

File: charts/ci-operator/templates/steps/standard.yaml [hunk 22]
```diff
@@ -86,6 +86,300 @@ step:
...
+params:
+  - name: TARGET_PORT
+    description: defines the entry port
+    type: string
+step:
+  name: nmap-vuln-standard
+  image: "instrumentisto/nmap"
+  script: |
+    mkdir -p $(workspaces.artifacts.path)/tests/nmap
+    echo y | apk add --no-cache git
```
Comment (david972) [inline charts/ci-operator/templates/steps/standard.yaml:261]: not available in CI

File: charts/ci-operator/templates/steps/standard.yaml [hunk 23]
```diff
@@ -86,6 +86,300 @@ step:
...
+  - name: TARGET_PATHS
+    description: define by client  all path targeted
+    type: string
+    default: "/"
+step:
+  name: nmap-vulners-standard
+  image: "instrumentisto/nmap"
+  script: |
+    mkdir -p $(workspaces.artifacts.path)/tests/nmap
+    echo y | apk add --no-cache git
```
Comment (david972) [inline charts/ci-operator/templates/steps/standard.yaml:287]: not available in CI

File: charts/ci-operator/templates/steps/standard.yaml [hunk 24]
```diff
@@ -86,6 +86,300 @@ step:
...
+params:
+  - name: TARGET_PORT
+    description: define by client the entry port
+    type: string
+step:
+  name: nmap-vulscan-standard
+  image: "instrumentisto/nmap"
+  script: |
+    mkdir -p $(workspaces.artifacts.path)/tests/nmap
+    echo y | apk add --no-cache git
```
Comment (david972) [inline charts/ci-operator/templates/steps/standard.yaml:310]: not available in CI

File: charts/ci-operator/templates/steps/standard.yaml [hunk 25]
```diff
@@ -86,6 +86,300 @@ step:
...
+      default: ""
+step:
+  name: build-quasar-standard
+  image: "node"
+  script: |
+    mkdir -p $(workspaces.artifacts.path)/builds/quasar
+    npm i -g @quasar/cli
+    cd $(resources.inputs.source.path)/$(params.PATH)
+    quasar build -m $(params.TYPE) $(params.EXTRA_PARAMETERS) | tee output
+    tar -czvf $(workspaces.artifacts.path)/builds/quasar/dist.tar.gz dist/*
```
Comment (david972) [inline charts/ci-operator/templates/steps/standard.yaml:381]: maybe in dockerfile so no step needed

Comment (jcrosnierw6d) [reply charts/ci-operator/templates/steps/standard.yaml:381]: Ok fine

File: charts/ci-operator/templates/steps/standard.yaml [hunk 26]
```diff
@@ -86,6 +86,300 @@ step:
...
+metadata:
+  name: git-leak-standard
+  labels:
+  {{- include "ci-operator.labels" . | nindent 4 }}
+  annotations:
+    ci.w6d.io/kind: generic
+    ci.w6d.io/task: git-leak
+    ci.w6d.io/order: "0"
+step:
+  name: git-leak-standard
```
[the remark this answers is not in the document]
Comment (jcrosnierw6d) [reply charts/ci-operator/templates/steps/standard.yaml]: done

File: charts/ci-operator/templates/steps/standard.yaml [hunk 27]
```diff
@@ -86,6 +86,300 @@ step:
...
 ---
+apiVersion: ci.w6d.io/v1alpha1
+kind: Step
+metadata:
+  name: git-leak-standard
+  labels:
+  {{- include "ci-operator.labels" . | nindent 4 }}
+  annotations:
+    ci.w6d.io/kind: generic
+    ci.w6d.io/task: git-leak
```
[the remark this answers is not in the document]
Comment (jcrosnierw6d) [reply charts/ci-operator/templates/steps/standard.yaml:97]: done

File: charts/ci-operator/templates/steps/standard.yaml [hunk 28]
```diff
@@ -86,6 +86,300 @@ step:
...
+  name: git-leak-standard
+  image: "zricethezav/gitleaks:latest"
+  script: |
+    mkdir -p $(workspaces.artifacts.path)/tests/report-gitleaks
+    gitleaks --repo-path $(resources.inputs.source.path) -v --report=$(workspaces.artifacts.path)/tests/report-gitleaks/report-pathsecretsscan.json
+---
+apiVersion: ci.w6d.io/v1alpha1
+kind: Step
+metadata:
+  name: code-cov-standard
```
[the remark this answers is not in the document]
Comment (jcrosnierw6d) [reply charts/ci-operator/templates/steps/standard.yaml]: done

File: charts/ci-operator/templates/steps/standard.yaml [hunk 29]
```diff
@@ -86,6 +86,300 @@ step:
...
+  labels:
+  {{- include "ci-operator.labels" . | nindent 4 }}
+  annotations:
+    ci.w6d.io/kind: generic
+    ci.w6d.io/task: code-cov
```
[the remark this answers is not in the document]
Comment (jcrosnierw6d) [reply charts/ci-operator/templates/steps/standard.yaml]: done

File: charts/ci-operator/templates/steps/standard.yaml (hunk 19 above)
Comment (jcrosnierw6d) [reply charts/ci-operator/templates/steps/standard.yaml]: done

File: charts/ci-operator/templates/steps/standard.yaml (hunk 21 above)
Comment (jcrosnierw6d) [reply charts/ci-operator/templates/steps/standard.yaml]: done

File: charts/ci-operator/templates/steps/standard.yaml (hunk 22 above)
Comment (jcrosnierw6d) [reply charts/ci-operator/templates/steps/standard.yaml]: done

File: charts/ci-operator/templates/steps/standard.yaml (hunk 23 above)
Comment (jcrosnierw6d) [reply charts/ci-operator/templates/steps/standard.yaml]: done

File: charts/ci-operator/templates/steps/standard.yaml (hunk 24 above)
Comment (jcrosnierw6d) [reply charts/ci-operator/templates/steps/standard.yaml]: done

File: charts/ci-operator/templates/steps/standard.yaml [hunk 30]
```diff
@@ -72,6 +72,82 @@ step:
...
+apiVersion: ci.w6d.io/v1alpha1
+kind: Step
+metadata:
+  annotations:
+    ci.w6d.io/kind: generic
+    ci.w6d.io/order: "0"
+    ci.w6d.io/task: deploy
+  name: deploy-init-helm-step-generic
+step:
+  image: w6dio/kubectl:v1.1.3
```
Comment (david972) [inline charts/ci-operator/templates/steps/standard.yaml:118]: bad version

File: charts/ci-operator/templates/steps/standard.yaml [hunk 31]
```diff
@@ -86,6 +162,237 @@ step:
...
+apiVersion: ci.w6d.io/v1alpha1
+kind: Step
+metadata:
+  annotations:
+    ci.w6d.io/kind: generic
+    ci.w6d.io/order: "5"
+    ci.w6d.io/task: nmap-nse
+  name: nmap-nse-clean-step-generic
+step:
+  image: w6dio/kubectl:v1.1.3
```
Comment (david972) [inline charts/ci-operator/templates/steps/standard.yaml:271]: wrong tag

File: charts/ci-operator/templates/steps/standard.yaml [hunk 32]
```diff
@@ -86,6 +162,237 @@ step:
...
+apiVersion: ci.w6d.io/v1alpha1
+kind: Step
+metadata:
+  annotations:
+    ci.w6d.io/kind: generic
+    ci.w6d.io/order: "5"
+    ci.w6d.io/task: owasp
+  name: owasp-clean-step-generic
+step:
+  image: w6dio/kubectl:v1.1.3
```
Comment (david972) [inline charts/ci-operator/templates/steps/standard.yaml:285]: wrong tag

File: charts/ci-operator/templates/steps/standard.yaml [hunk 33]
```diff
@@ -86,6 +162,237 @@ step:
...
+apiVersion: ci.w6d.io/v1alpha1
+kind: Step
+metadata:
+  annotations:
+    ci.w6d.io/kind: generic
+    ci.w6d.io/order: "1"
+    ci.w6d.io/task: nmap-nse
+  name: nmap-nse-deploy-step-generic
+step:
+  image: w6dio/kubectl:v1.1.3
```
Comment (david972) [inline charts/ci-operator/templates/steps/standard.yaml:327]: wrong tag

File: charts/ci-operator/templates/steps/standard.yaml [hunk 34]
```diff
@@ -86,6 +162,237 @@ step:
...
+apiVersion: ci.w6d.io/v1alpha1
+kind: Step
+metadata:
+  annotations:
+    ci.w6d.io/kind: generic
+    ci.w6d.io/order: "1"
+    ci.w6d.io/task: owasp
+  name: owasp-deploy-step-generic
+step:
+  image: w6dio/kubectl:v1.1.3
```
Comment (david972) [inline charts/ci-operator/templates/steps/standard.yaml:343]: wrong tag

**Pull request merged**